admin/pre_reg/views.py

```diff
@@ -20,169 +17,172 @@
-    """
-    return user.is_in_group('prereg_group')
-
-
-@user_passes_test(is_in_prereg_group)
-def prereg(request):
-    """Redirects to prereg page if user has prereg access
-    :param request: Current logged in user
-    :return: Redirect to prereg page with username, reviewers, and user obj
-    """
-    paginator = Paginator(get_prereg_drafts(user=request.user), 5)
-
-    try:
-        page_number = int(request.GET.get('page'))
-    except (TypeError, ValueError):
-        page_number = 1
-
-    page = paginator.page(page_number)
-
-    try:
-        drafts = [serializers.serialize_draft_registration(d, json_safe=False) for d in page]
-    except EmptyPage:
-        drafts = []
-
-    for draft in drafts:
-        draft['form'] = DraftRegistrationForm(draft)
-
-    context = {
-        'drafts': drafts,
-        'page': page,
-        'IMMEDIATE': serializers.IMMEDIATE,
-    }
-    return render(request, 'pre_reg/prereg.html', context)
-
-
-@user_passes_test(is_in_prereg_group)
-def view_draft(request, draft_pk):
-    """Redirects to prereg form review page if user has prereg access
-    :param draft_pk: Unique id for selected draft
-    :return: Redirect to prereg form review page with draft obj
-    """
-    draft = get_draft_or_error(draft_pk)
-    context = {
-        'draft': serializers.serialize_draft_registration(draft)
-    }
-    return render(request, 'pre_reg/edit_draft_registration.html', context)
-
-
-@user_passes_test(is_in_prereg_group)
-def view_file(request, node_id, provider, file_id):
-    file = FileNode.load(file_id)
-    wb_url = file.generate_waterbutler_url()
-    return redirect(wb_url)
-
+from admin.base.utils import PreregAdmin
+
+
+class DraftListView(PreregAdmin, ListView):
+    template_name = 'pre_reg/draft_list.html'
+    paginate_by = 10
+    paginate_orphans = 1
+    ordering = '-approval.initiation_date'
+    context_object_name = 'draft'
+
+    def get_queryset(self):
+        prereg_schema = MetaSchema.find_one(
+            Q('name', 'eq', 'Prereg Challenge') &
+            Q('schema_version', 'eq', 2)
+        )
+        query = (
+            Q('registration_schema', 'eq', prereg_schema) &
+            Q('approval', 'ne', None)
+        )
+        return DraftRegistration.find(query).sort(self.ordering)
+
+    def get_context_data(self, **kwargs):
+        query_set = kwargs.pop('object_list', self.object_list)
+        page_size = self.get_paginate_by(query_set)
+        paginator, page, query_set, is_paginated = self.paginate_queryset(
+            query_set, page_size)
+        return {
+            'drafts': [
+                serializers.serialize_draft_registration(d, json_safe=False)
+                for d in query_set
+            ],
+            'page': page,
+        }
+
+
+class DraftDetailView(PreregAdmin, DetailView):
+    template_name = 'pre_reg/draft_detail.html'
+    context_object_name = 'draft'
+
+    def get(self, request, *args, **kwargs):
+        try:
```

Comment:
You can delete this method and just handle the exceptions you think might happen by raising the proper HttpXXX exception.